clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -334,6 +334,13 @@ public OffsetResetStrategy resetStrategy(TopicPartition partition) {
         return assignedState(partition).resetStrategy;
     }
 
+    public boolean hasAllFetchPositions(Collection<TopicPartition> partitions) {
+        for (TopicPartition partition : partitions)
+            if (!hasValidPosition(partition))
+                return false;
+        return true;
+    }
+
     public boolean hasAllFetchPositions() {
```

Comment:
I submitted an update that refactors the original method. I'd be happy to take your second suggestion too if you think that's a better way to go. Thanks.